File: LeetCode/Medium/0137-single-number-ii/0137-single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        li = [0]*32

        for n in nums:
            for i in range(32):
                if (n >> i) & 1:
                    li[i] += 1

        res = 0        
        for i in range(len(li)):
            if li[i]%3 != 0:
                res |= 1<<i

        if (res>>31) & 1:
            res -= (1<<32)
    
        return res

File: LeetCode/Medium/0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int n = nums.size();

        int arr[32] = {0};

        for(int i=0; i<n; i++)
            for(int j=0; j<32;j++)
                if(nums[i]>>j & 1) arr[j]++;
            
        int res = 0;

        for(int i=0; i<32; i++)
            if(arr[i]%3) res |= 1<<i;

        return res;
    }
};